src/runtime/LegacyStory.svelte

```diff
@@ -6,22 +6,21 @@
    * Vite pre-transform hook does codemod where this component gets transformed into new `Story` component destrucutred from `defineMeta`.
    */
   import type { Component, ComponentProps } from 'svelte';
-  import type { EmptyObject } from 'type-fest';
 
   import type Story from './Story.svelte';
 
-  import { type StoryRendererContext } from '#runtime/contexts/renderer.svelte';
-  import type { Meta } from '#types';
+  import { type StoryRendererContext } from './contexts/renderer.svelte';
 
-  type Props = ComponentProps<Story<EmptyObject, Component, Meta<Component>>> & {
+  type $$Props = ComponentProps<Story<Component>> & {
     /**
      * Which one of `<Template>` id should be used for rendering this story children?
+     * @deprecated use Svelte [snippets]() instead.
+     * @see
      */
     template?: string | null;
   };
 
-  let { args }: Props = $props();
-
+  let args: $$Props['args'];
```

Comment:
I don't understand this change TBH

Reply:
Apparently Svelte language tools are so smart and capable to recognize when legacy API is used.
And producing two different types (via import `Story|LegacyStory.svelte`.

While using `LegacyStory`, the type of this component was treated as modern one (`svelte@5` - `Component` - function).
Hence props types we're mismatched and couldn't use let:*` directives without type errors.
Examples of these errors can be found in the other [PR comment](https://github.com/storybookjs/addon-svelte-csf/pull/215#issuecomment-2418701326)  (which can be closed if we merge this one)


So, for language tools to treat this component as legacy `SvelteComponent` (class), I had to use the legacy syntax for defining props.

Reply:
oh damn that's good. great find! ❤️